static/js/tools/map/chart_loader.tsx

```diff
@@ -581,19 +579,12 @@ function fetchData(
         if (enclosedPlacesPop && enclosedPlacesPop.facets) {
           Object.assign(metadataMap, enclosedPlacesPop.facets);
         }
-        const sampleDates =
-          placeStatDateWithinPlace.data[statVar.dcid].statDate && showTimeSlider
-            ? getTimeSliderDates(
-                metadataMap,
-                placeStatDateWithinPlace.data[statVar.dcid].statDate
-              )
-            : {};
         let legendBounds: Record<string, [number, number, number]> = {};
-        if (BEST_AVAILABLE_METAHASH in sampleDates) {
+        if (BEST_AVAILABLE_METAHASH in allSampleDates) {
           // Set dates for "Best Available" to best series
-          const bestAvailableHash = sampleDates[BEST_AVAILABLE_METAHASH][0];
-          sampleDates[BEST_AVAILABLE_METAHASH] =
-            sampleDates[sampleDates[BEST_AVAILABLE_METAHASH][0]];
+          const bestAvailableHash = allSampleDates[BEST_AVAILABLE_METAHASH][0];
+          allSampleDates[BEST_AVAILABLE_METAHASH] =
+            allSampleDates[allSampleDates[BEST_AVAILABLE_METAHASH][0]];
```

Comment:
instead of this logic here, can we just have allSampleDates[BEST_AVAILABLE_METAHASH] populated with the correct list of dates in the compute hook? Or is there a reason it needs to be populated here?

Reply:
I have not dig deep into the logic here but just to keep the logic matching. In next step where legend bound is computed, let me clean this up.

Reply:
I think it can be computed in the hook - sorry I'm not sure why I did it like this before

Reply:
okay actually thinking about it more, I think the reason it's like this was because we wanted to know which facet we're choosing as the "best available"
without time slider, the "best available" view could potentially draw from multiple facets and dates at once, but now since we're forcing a date, I think we had decided to just show a single facet (whichever had the best coverage of observations). so this gives us access to which facet to display. this is probably not the neatest way to do this though, so makes sense to clean this up

Reply:
Sounds good to keep it as is for this PR and revisit this in a future PR!